src/server/gc_worker.rs

```diff
@@ -212,6 +216,53 @@ impl<E: Engine> GcRunner<E> {
         .map_err(Error::from)
     }
 
+    /// Check need gc without getting snapshot.
+    /// If this is not supported or any error happens, returns true to do further check after
+    /// getting snapshot.
+    fn need_gc(&self, ctx: &Context, safe_point: TimeStamp) -> bool {
+        let region_info_accessor = match &self.region_info_accessor {
+            Some(r) => r,
+            None => return true,
+        };
+
+        let db = match &self.local_storage {
+            Some(db) => db,
+            None => return true,
+        };
+
+        let (tx, rx) = mpsc::channel();
```

Comment:
How about `paired_future_callback'?

Reply:
Here we don't need a future. I think the original `mpsc` is good.

Reply:
emmmm should I change it back..

Reply:
It's not very important, but I'm personally inclined to change it back.